test/unit-tests/fiber.mlt

```diff
@@ -0,0 +1,163 @@
+open Dune;;
+open Stdune;;
+open Fiber;;
+open Fiber.O;;
+
+let failing_fiber () =
+  Fiber.yield ()
+  >>= fun () ->
+  raise Exit
+;;
+
+let long_running_fiber () =
+  let rec loop n =
+    if n = 0 then
+      Fiber.return ()
+    else
+      Fiber.yield ()
+      >>= fun () ->
+      loop (n - 1)
+  in
+  loop 10
+;;
+
+let ivar : unit Fiber.Ivar.t = Fiber.Ivar.create ();;
+
+let getter_fiber () =
```

Comment:
Let's use `Fiber.never` instead of `getter_fiber ()`. It will be clearer that we are using a fiber that never terminates